calblueprint/girls-write-now
# [auth] Date picker crossplatform

[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- Added a date picker modal in the settings and onboarding page for users to select their birthday 
- Warning pops up after user has set their birthday
- Date picker is only accessible if user hasnt already set their birthday, disabled after user has set their birthday

## Relevant Links

### Notion Sprint Task

[//]: # 'Add the relevant Notion link(s) here'
https://www.notion.so/calblueprint/auth-Date-picker-crossplatform-c3e6ced7d28344209c1bbd130ee5ee5e
### Online sources

[//]: # 'Optional - copy links to any tutorial or documentation that was useful to you when working on this PR'
https://github.com/mmazzarolo/react-native-modal-datetime-picker

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review

[//]: # 'The order in which to review files and what to expect when testing locally'

- Create a new account or use an account that doesn't have a birthday set yet
- Set the birthday either in the onboarding page or in the settings page 
- Check that after saving edits, you can no longer set your birthday
- Sign out and check if your birthday has been set

## Next steps

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases

[//]: # 'Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally'

- Update profile doesn't seem to be working, setting my pronouns/race/gender/birthday doesn't seem to stick after saving edits then signing out and signing back in 

### Screenshots

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

https://github.com/calblueprint/girls-write-now/assets/156966341/9372eb25-200e-441c-8b54-77beb571631a


https://github.com/calblueprint/girls-write-now/assets/156966341/1ee11441-f772-4250-9715-bdd6f29a98d4



CC: @adityapawar1

[//]: # 'This tags in Adi as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
